logstash-core/src/main/java/org/logstash/ext/JrubyMemoryWriteClientExt.java

```diff
@@ -43,29 +46,37 @@ public JrubyMemoryWriteClientExt(final Ruby runtime, final RubyClass metaClass)
     }
 
     private JrubyMemoryWriteClientExt(final Ruby runtime, final RubyClass metaClass,
-        final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue) {
+                                      final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue) {
         super(runtime, metaClass);
         this.queue = queue;
     }
 
     public static JrubyMemoryWriteClientExt create(
-        final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue) {
+            final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue) {
         return new JrubyMemoryWriteClientExt(RubyUtil.RUBY,
-            RubyUtil.MEMORY_WRITE_CLIENT_CLASS, queue);
+                RubyUtil.MEMORY_WRITE_CLIENT_CLASS, queue);
     }
 
     @Override
     protected JRubyAbstractQueueWriteClientExt doPush(final ThreadContext context,
-        final JrubyEventExtLibrary.RubyEvent event)
-        throws InterruptedException {
```

Comment:
Pulling the `throws InterrutedException` off of `JRubyAbstractQueueWriteClientExt#doPush` (and similar methods) greatly expands the scope of possible effect. I am _very_ hesitant to recast interruptions as runtime issues, as interruptions are most often caused by a shutdown in progress and determining the net effect could be a monumental effort. A quick look at the underlying queue implementation for the memory queue indicates that the close operation is a no-op and theoretically _shouldn't_ result in interruptions, but I haven't also validated the pipeline shutdown path to see if stalled threads can also be interrupted from the outside.

Reply:
I have pulled off of introducing `QueueRuntimeException` exception on queue operations, including wrapping `InterruptedException`s. The reason, I am not 100% confident if clients/callers (especially ruby plugin sources) properly handling the exceptions, especially if they intentionally expect the `InterruptedException`.

Stepping back to origin `"Tried to write to a closed queue." IllegalStateException` issue, replacing `IllegalStateException` with `QueueRuntimeException` in specific locations such as `JrubyAckedWriteClient#push()`, `Queue#constructor()` or raising it when queue closed seems to me a meaningful change.